src/components/AssignRole/AssignRole.tsx

```diff
@@ -0,0 +1,28 @@
+import Modal from '@/components/Modal/Modal'
+import { useModal } from '@/contexts/ModalContext'
+import { Button } from '@/components/Button/Button'
+
+import styles from './AssignRole.module.scss'
+import modalStyles from '@/components/Modal/Modal.module.scss'
+
+export default function AssignRole({ profileId }: { profileId: string }) {
+  const { isModalVisible, closeModal } = useModal()
+
+  if (!isModalVisible) return null
```

Comment:
**nitpick:** I think we don't need this `if` now. We can decouple these modal, and call them eg. `AssignRoleModal`, and pass them eg. `onClose`. It will be easier to manage, and test them